bench/workgen/workgen.cxx

```diff
@@ -431,9 +431,10 @@ ContextInternal::~ContextInternal()
 }
 
 int
-ContextInternal::create_all()
+ContextInternal::create_all(std::recursive_mutex &mutex)
 {
     if (_runtime_alloced < _tint_last) {
+        const std::lock_guard<std::recursive_mutex> lock(mutex);
         // The array references are 1-based, we'll waste one entry.
         TableRuntime *new_table_runtime = new TableRuntime[_tint_last + 1];
```

Comment:
Are we allocating a new array each time we create a table (once the count of the tables is greater than runtime alloced)? We could allocate double the size of the array and only expand once we reach the limit, or we can follow a different allocation strategy.

Reply:
Maybe _table_runtime could be a std::vector, which does preallocation?  Still should be locked when accessing or adding.